certification/internal/shell/has_license.go

```diff
@@ -1,30 +1,43 @@
 package shell
 
 import (
-	"os/exec"
 	"strings"
 
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/cli"
 	log "github.com/sirupsen/logrus"
 )
 
 type HasLicenseCheck struct{}
 
 func (p *HasLicenseCheck) Validate(image string) (bool, error) {
-	stdouterr, err := exec.Command("podman", "run", "-it", "--rm", "--entrypoint", "ls", image, "-A", "/licenses").CombinedOutput()
-	result := string(stdouterr)
+
+	return p.validate(image)
+}
+
+func (p *HasLicenseCheck) validate(image string) (bool, error) {
+	runOpts := cli.ImageRunOptions{
+		EntryPoint:     "ls",
+		EntryPointArgs: []string{"-A", "/licenses"},
+		LogLevel:       "debug",
+		Image:          image,
+	}
+	runReport, err := podmanEngine.Run(runOpts)
+	line := runReport.Stdout
 	if err != nil {
-		if strings.Contains(result, "No such file or directory") || result == "" {
+
+		if strings.Contains(line, "No such file or directory") || line == "" {
 			log.Warn("license not found in the container image at /licenses")
 			return false, nil
 		}
 
-		log.Error("some error attempting to identify if /licenses container the license: ", err)
+		log.Error("some error attempting to identify if /licenses cotainer the license: ", err)
+		log.Debugf("Stdout: %s", runReport.Stdout)
+		log.Debugf("Stderr: %s", runReport.Stderr)
 		return false, err
 	}
 
-	// sanity check - in case we don't get an error, but also don't have the file.
-	if strings.Contains(result, "No such file or directory") || result == "" {
+	if strings.Contains(line, "No such file or directory") || line == "" {
 		log.Warn("license not found in the container image at /licenses")
```

Comment:
could this message be part of the `HelpText`? If so, you can replace this if block with
`return  !strings.Contains(line, "No such file or directory") && line != "", nil`